src/parent/reducers/reducers.js

```diff
@@ -33,30 +35,32 @@ const rootReducer = reduceReducers(
         }
 
         case ACTION_TYPES.CLOSE: {
-            const newClosedWindow = Object.assign({}, state.childWindows[action.windowName], {
-                date: action.date
-            });
-
-            const newChildWindows = Object.assign({}, state.childWindows);
-
-            delete newChildWindows[action.windowName];
+            // FIXME Solve why close event is fired twice when re-opened window is closed! Event listener not removed?
+            if (!state.childWindows[action.windowName]) return state;
```

Comment:
This is the workaround; the rest is code sanitising and refactoring into ES syntax.